src/anyio/_core/_sockets.py

```diff
@@ -186,6 +186,15 @@ async def try_connect(remote_host: str, event: Event) -> None:
     try:
         addr_obj = ip_address(remote_host)
     except ValueError:
+        addr_obj = None
+
+    if addr_obj is not None:
+        if isinstance(addr_obj, IPv6Address):
+            target_addrs = [(socket.AF_INET6, addr_obj.compressed)]
+        else:
+            target_addrs = [(socket.AF_INET, addr_obj.compressed)]
+
+    else:
```

Comment:
this might be nicer as:

```python
if isinstance(addr_obj, IPv6Address):
    ...
elif isinstance(addr_obj, IPv4Address):
    ...
elif addr_obj is None:
    ...
else:
    assert_never(addr_obj)
```

Reply:
agree, and imagine how nice it'll be when we can use `match`/`case` :star_struck:

Reply:
Feel free to do that, but if you do, start with `if addr_obj is None`, as it's the cheapest check. And finish with `else: ... # process for ipv4`.